### ISSUE TITLE
NaNpx appears in attributes

### ISSUE THREAD

**Issue by lfrohman** [angular-ui/ui-grid#4795]
It appears that the min-width and min-height attributes are getting a javascript NaN.
This bug has happened a few times, but it is very rare.
    "angular": "~1.2.16",
    "angular-ui": "0.4.0",
    "angular-ui-grid": "~3.0.7",
. . .
attached are the relevant html and a screen print.

![ng_grid_bug](https://cloud.githubusercontent.com/assets/1037104/11511201/8e32fe86-981c-11e5-9fe6-5a6795fdd71b.png)

---

[ng_grid_bug.html.txt](https://github.com/angular-ui/ui-grid/files/48934/ng_grid_bug.html.txt)

**Comment by 314rated** [comment]
What is the solution to this problem? Even I am facing the same issue.
Also, I need to use a custom css with ui-grid. So should resorting to smart table or any other angular table be a better option?
Thanks

**Comment by swalters** [comment]
Would be great is we had some simple steps to recreate and maybe a plunkr.

**Comment by lfrohman** [comment]
Yes, that would be great. Sorry, but this bug occurs very rarely, and I
could never recreate
the steps to reproduce it.

**Comment by jamesbrucepower** [comment]
I had the same issue, when I debugged it, I found that the calculations on min-width are done on the drawnWidth in the 

GridColumn.prototype.getColClassDefinition method

and in my case I was hiding the ui-grid when the page loaded, meaning that the drawnWidth wasn't yet calculated which resulted in a NaN

Not sure if there is a way around this though?

**Comment by jacques-hoventer** [comment]
Not sure why this was closed because I'm getting the exact same behavior and there's no solution here?

**Comment by PiDayDev** [comment]
Occasionally getting the same behaviour with AngularJS v1.6.9 + ui-grid v4.4.7
![uigrid](https://user-images.githubusercontent.com/15138318/39574711-74a75f3c-4ed8-11e8-93db-c2b43d2cc460.png)

**Comment by jacques-hoventer** [comment]
@PiDayDev I was using ngShow and switched it out for ngIf which solved the problem for me, but it's obviously not a proper fix.

**Comment by mportuga** [comment]
Sorry, there were no reproduction steps or activity in this bug, so I had to close it.

**Comment by jacques-hoventer** [comment]
@mportuga I actually found this issue #3440 which is where I came across both a plnkr reproduction of the issue as well as the ngIf fix.

**Comment by PiDayDev** [comment]
@jacques-hoventer in my case, it's on an element without `ng-show` or `ng-if`, even tough its container is initially hidden. I'm trying to delay the element displaying, or to force a resize/refresh event, since both workarounds seem to help

**Comment by orneryd** [comment]
Thanks for contributing! This repository has been transferred, and we're preparing the new 5.x release. We're closing older issues and pull requests as part of the reset. Please join us on Discord to follow the work or continue the conversation: https://discord.gg/Baz4w8ZWN
